modules/json/filterx-cache-json-file.c

```diff
@@ -209,36 +250,75 @@ _deep_freeze(FilterXFuntionCacheJsonFile *self, FilterXObject *object)
 
   object = filterx_ref_unwrap_ro(object);
   if (filterx_object_is_type(object, &FILTERX_TYPE_NAME(json_object)))
-    _deep_freeze_dict(self, object);
+    _deep_freeze_dict(self, object, root_object);
 
   if (filterx_object_is_type(object, &FILTERX_TYPE_NAME(json_array)))
-    _deep_freeze_list(self, object);
+    _deep_freeze_list(self, object, root_object);
+}
+
+gboolean
+_load_json_file_version(FilterXFunctionCacheJsonFile *self, GError **error)
+{
+  main_loop_assert_main_thread();
+  if (self->history_index >= FROZEN_OBJECTS_HISTORY_SIZE)
+    {
+      main_loop_reload_config(main_loop_get_instance());
```

Comment:
Can we move out this condition and the reload call to `_file_monitor_callback` (the name `_load_json_file_version` does not suggest that such intrusive side-effect might happen)?
Once it is done, can we add a comment to `_file_monitor_callback()` that this function may reload the configuration, so nothing complicated should be done with config-related references on the caller side?